core/classes/Widgets/ProfileWidgetBase.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+abstract class ProfileWidgetBase implements Widget {
```

Comment:
Don't totally love the design of this system, but at the same time, a new DB table for profile widgets seems overkill - and needing to repeat a bunch of logic doesn't seem nice either...